File: src/java/WebNinja/src/test/java/webninja/selenuim/HomePage.java
package webninja.selenuim;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import webninja.UserRepository;

import java.util.List;

public class HomePage {
    private final WebDriver webDriver;
    private final CodeTrack codeTrack;
    private final UserRepository userRepository;

    public HomePage(WebDriver webDriver, CodeTrack codeTrack, UserRepository userRepository) {
        this.webDriver = webDriver;
        this.codeTrack = codeTrack;
        this.userRepository = userRepository;
    }

    public IssuePage showIssueWithTitle(String issueTitle) {
        List<WebElement> rows = webDriver.findElements(By.xpath("//table[@id='results']/tbody/tr"));
        for (WebElement row : rows) {
            WebElement summary = row.findElement(By.xpath("/td[4]"));
            if (summary.getText().equals(issueTitle)) {
                row.findElement(By.xpath("/td[1]/a")).click();
                return new IssuePage(webDriver, codeTrack, userRepository);
            }
        }
        throw new IllegalArgumentException("No issue found for title" + issueTitle);
    }
}

File: src/dotnet/WebNinja/WebNinja/selenium/HomePage.cs
using System.Collections.ObjectModel;
using OpenQA.Selenium;
using WebNinja.webninja;

namespace WebNinja.selenium
{
    public class HomePage : PageObject
    {
        private readonly UserRepository _userRepository;

        public HomePage(IWebDriver driver, UserRepository userRepository) : base(driver)
        {
            _userRepository = userRepository;
        }

        public string ProjectName
        {
            set { SelectOption(By.Name("project"), value); }
        }

        public IssuePage ShowIssueWithTitle(string title)
        {
            ReadOnlyCollection<IWebElement> rows = Driver.FindElements(By.XPath("//table[@id='results']/tbody/tr"));
            foreach (IWebElement row in rows)
            {
                IWebElement summary = row.FindElement(By.XPath("td[4]"));
                if (summary.Text.Equals(title))
                {
                    row.FindElement(By.XPath("td[1]/a")).Click();
                    return new IssuePage(Driver, _userRepository);
                }
            }
            throw new NoSuchElementException("No issue found for title" + title);
        }
    }
}